### ISSUE TITLE
Album list performance

### ISSUE THREAD

**Issue by mtk64** [Rello/audioplayer#130]
### Expected behavior
Albums should get visible within seconds, also on low performant clients

### Actual behavior
Albums get visible within one or two minutes 

### Steps to reproduce the behavior
1. Open audioplayer on a Raspberry Pi (mine is 3)
2. Wait until albums become visible in Audioplayer, wheel is spinning

### Possible solution (optional)
Maybe use more performance on server itself? 
Displaying some hundred images of that size should be possible much faster.
I can see heavy load on one of the 4 cpus working on the pi. 
Seems to do a lot on client cpu ... maybe use more cpus ?

**ownCloud/Nextcloud version:** (see `/status.php`)
11.1

**Updated from an older ownCloud/Nextcloud or fresh install:**
Fresh

**Audio Player version:** (see `/settings/apps`)
actual

**Where did you install Audio Player from:**
OC Store

**Are you using external storage, if yes which one:** `local/smb/sftp/...`
local with nfs mount and smb mounts as well

### Client configuration
**Operating system:**
Raspbian with Pixel Desktop

**Browser:**
Chrome

**Comment by Rello** [comment]
Hello,

are you running the NC-Server + The Browser (Audio Player) on the same RPi3?
because then you would have **4** loads:
- NC/Audio Player php processes
- Webserver (nginx? apache?)
- Database selects
- Pixcel Desktop

which one is the single process with heavy load?

for a validation of your bottleneck it would be useful to open Audio Player on a separate box for comparison.
also note that after the first start, the images will be cached.

About how many albums are we talking about?

**Comment by mtk64** [comment]
- No, NC is running on a VM with Apache2 and MySQL on my XEN Server.

- The process is only chromium browser (renderer) with 120%

- Checked that with 50  and 800 Albums as well. 
 
Performance on my Windows PC is ok, needs just 15 seconds to start displaying around 13.000 titles in their albums, Dont know how many albums there are exactly. :-)

**Comment by mtk64** [comment]
There are 700 albums, and here the results from testing. For comparison taken on 2 clients with much less power than a pc.

1. it takes 25 seconds on ipad air to show album cover arts  
2. it takes more than a minute on rpi3 with chromium browser to show album cover arts  

Server is quad core intel machine with a xen vm and a fresh nc11.1 installation
(daily) Build:2017-01-15  16:04:22
Apache 2
PHP Version: 5.6.29
Memory Limit: 512.0 MB
Database Type: mysql Version: 10.0.29 Size: 57.4 MB

**Comment by Rello** [comment]
@mtk64 would it be possible to get a test-account on your installation to which you share these albums?
I don´t have the possibility to test with these many albums but I would like to debug some frontend parts.

**Comment by Rello** [comment]
I could reproduce the issue.
currently /getMusic is transferring the metadata for all Albums+Tracks plus all Coverarts in a base64 string in one json ajax.

on the upper environment, this is around 12MB which takes time to progress.

An enhancement is in progress to separate the content.

- Covers will be displayed as real images to take advantage of parallel transfer and caching
- The metadata will be reduced and also gzipped where supportd

**Comment by Rello** [comment]
#149
